app/containers/Questions/Header.jsx

```diff
@@ -160,42 +274,94 @@ export const Header = ({
-      </PageContentHeader>
-
-      {communityEditingAllowed && (
-        <button onClick={routeToEditCommunity} className="df aic mt12">
-          <IconMd icon={pencilIcon} color={colors.btnColor || TEXT_PRIMARY} />
-          <Span className="ml-1" color={colors.btnColor || TEXT_PRIMARY}>
-            {t('common.editCommunity')}
-          </Span>
-        </button>
-      )}
+      <PageContentHeaderContainer>
+        <PageContentHeader className="d-flex align-items-center">
+          <CommunitySelector
+            isArrowed
+            Button={Button}
+            toggle={(choice) => {
+              createdHistory.push(routes[route](choice, false, false));
+              setTypeFilter(choice);
+            }}
+            showOnlyFollowed={isFeed}
+            selectedCommunityId={communityIdFilter}
+            communities={communities}
+          />
+          {/* PEER-451: Hide Subscribe button from single community mode
+          {!!displaySubscribeButton && (
+            <PageContentHeaderRightPanel
+              className={`right-panel m-0 ml-${single ? 3 : 4}`}
+            >
+              <FollowCommunityButton
+                communityIdFilter={single || communityIdFilter}
+                followedCommunities={followedCommunities}
+              />
+            </PageContentHeaderRightPanel>
+          )} */}
+        </PageContentHeader>
+        {communityEditingAllowed && (
+          <EditCommunityButton>
+            <button onClick={routeToEditCommunity} className="aic">
+              <IconMd
+                icon={pencilIcon}
+                color={colors.btnColor || TEXT_PRIMARY}
+              />
+              <Span className="ml-1" color={colors.btnColor || TEXT_PRIMARY}>
+                {t('common.editCommunity')}
+              </Span>
+            </button>
+          </EditCommunityButton>
+        )}
+        {Boolean(tags.length) && (
+          <TagFilterContainer>
+            <ScrollContainer>
+              <div className="df mt-md-3 mt-sm-0">
+                {tags.map((tag) => {
+                  const removeTagFilter = (removedTag) => {
+                    const searchParams = new URLSearchParams(
+                      createdHistory.location.search,
+                    );
+                    const searchParamsTags = searchParams.get('tags');
+                    const allTags = searchParamsTags?.split(':');
+                    const result = allTags?.filter(
+                      (item) => item !== removedTag,
+                    );
+                    if (result?.length) {
+                      searchParams.set('tags', result.join(':'));
+                    } else {
+                      searchParams.delete('tags');
+                    }
+                    createdHistory.push(
+                      `${createdHistory.location.pathname}?${searchParams}`,
+                    );
+                  };
+                  return (
+                    <TagFilter key={tag}>
```

Comment:
Extract this item into a separate component.

Reply:
+